### PR TITLE
[AKS PREVIEW]aks create: Change the default value of --enable-msi-auth-for-monitoring to true and add check for airgap clouds

### PR DESCRIPTION
**Pull request by wanlonghenry**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS PREVIEW

**Comment by wanlonghenry** [comment]
cli change: https://github.com/Azure/azure-cli/pull/26356

**Comment by wanlonghenry** [comment]
aks enable addons: 
image

**Comment by wanlonghenry** [comment]
image

**Comment by wanlonghenry** [comment]
image

File: src/aks-preview/HISTORY.rst [hunk 1]
```diff
@@ -15,6 +15,7 @@ Pending
 
 0.5.139
 * `az aks create` and `az aks nodepool add`: Add warning message when specifying `--os-sku` to `Mariner` or `CBLMariner`.
+* `az aks create` and `az aks enable-addons`: Change the default value of `--enable-msi-auth-for-monitoring` to `true` and add check for airgap clouds
```
Comment (ganga1980) [inline src/aks-preview/HISTORY.rst:18]: update to this include **monitoring**  since this only applicable for monitoring addon

Comment (wanlonghenry) [reply src/aks-preview/HISTORY.rst]: changed

**Comment by wanlonghenry** [comment]
image

**Comment by zhoxing-ms** [comment]
Do you want to release a new extension for this PR change? If so, please upgrade the version defined in the `setup.py`

File: src/aks-preview/HISTORY.rst [hunk 2]
```diff
@@ -15,6 +15,7 @@ Pending
 
 0.5.139
 * `az aks create` and `az aks nodepool add`: Add warning message when specifying `--os-sku` to `Mariner` or `CBLMariner`.
+* `az aks create` and `az aks enable-addons`: Change the default value of `--enable-msi-auth-for-monitoring` to `true` and add check for airgap clouds for monitoring addon
```
Comment (FumingZhang) [inline src/aks-preview/HISTORY.rst:18]: Would be better to release a new version of aks-preview. Please move this block several lines above, choose a new version number, and update the version number in setup.py accordingly.

Comment (wanlonghenry) [reply src/aks-preview/HISTORY.rst]: thank you for the suggestion, changed to use 0.5.140

**Review by ganga1980 — APPROVED** [review]

**Review by FumingZhang — APPROVED** [review]
LGTM

**Review by zhoxing-ms — APPROVED** [review]

**Comment by wanlonghenry** [comment]
the above issue fixed in: https://github.com/Azure/azure-cli/pull/26528
image

**Pull request merged**